mtags/src/main/scala/scala/meta/internal/mtags/MtagsEnrichments.scala

```diff
@@ -359,4 +363,32 @@ trait MtagsEnrichments {
     def isVar: Boolean = (properties & p.VAR.value) != 0
     def isVal: Boolean = (properties & p.VAL.value) != 0
   }
+
+  implicit class XtensionStream[A](stream: java.util.stream.Stream[A]) {
+    import scala.collection.JavaConverters._
+    def asScala: Generator[A] = {
+      Generator.selfClosing((stream.iterator.asScala, () => stream.close()))
+    }
+  }
+
+  implicit class XtensionAbsolutePath(path: AbsolutePath) {
+    def parent: AbsolutePath = {
+      AbsolutePath(path.toNIO.getParent)
+    }
+
+    def exists: Boolean = {
+      Files.exists(path.toNIO)
+    }
+
+    def iterate: Generator[AbsolutePath] = {
```

Comment:
I am not sure about the name. While `walk` is sometimes used in this context, I feel it does not carry the information of what is actually done. I was thinking about something with the `hierarchy` in the name, like `traverseHierarchy`, `toTree`, maybe even the closest-to-original `walkTree` ? What do you think?

Reply:
Hmm, how about `list` and `listRecursive`?